infra/network-isolation.bicep

```diff
@@ -18,6 +21,24 @@ resource appServicePlan 'Microsoft.Web/serverfarms@2022-03-01' existing = {
   name: appServicePlanName
 }
 
+module backendNsg './core/networking/nsg.bicep' = if (usePrivateEndpoint) {
```

Comment:
Can we use AVM's module here?
https://github.com/Azure/bicep-registry-modules/tree/main/avm/res/network/network-security-group

Reply:
Example- https://github.com/Azure/bicep-registry-modules/tree/main/avm/res/network/network-security-group#example-3-waf-aligned

Reply:
You'd have to define the securityrules as a variable in this file, in that case.

Reply:
But we can also defer the AVM til later, if you prefer, so that we can get this in sooner.

Reply:
Are there AVM docs? Didn't realize we have them now